service/catalog.go

```diff
@@ -84,16 +160,33 @@ func updateCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, e
 		return http.StatusBadRequest, err
 	}
 
-	if err := db.Model(&model.CatalogModel{}).Where(&model.CatalogModel{
+	vars := mux.Vars(r)
+	oldCatalogName := vars["catalog"]
+
+	oldCatalog := model.CatalogModel{
 		Catalog: model.Catalog{
-			Name:          catalogModel.Name,
+			Name:          oldCatalogName,
 			EnvironmentId: envId,
 		},
-	}).Update(catalogModel).Error; err != nil {
+	}
+
+	if isDuplicateGlobalName(catalogModel) {
```

Comment:
Why call isDuplicateGlobalName and isDublicateEnvName back-to-back when you have isDuplicateName which does both?